Wip/mehrad/kubeflow

Mehrad0711 opened the pull request:
Most changes in this PR are for SPL pipelines: fixing things and adding new features. 

Some of the images for thingtalk2 come with preinstalled genie-toolkit and thingtalk, and running rebuild fails. I changed Dockerfile to account for that but not sure if there is a better way instead of `if else`.

Mehrad0711 committed:
submit_job.py: add license

Mehrad0711 committed:
rebuild-image: don't remove lib.sh

Mehrad0711 committed:
make docker work for new images

Mehrad0711 committed:
Update pip

new version correctly resolve dependency conflics for torch and transformers

Mehrad0711 committed:
Add use_apex option for docker

Mehrad0711 committed:
Cleanup imports in pipelines

Mehrad0711 committed:
submit_job.sh: remove default version

not all pipelines use those

Mehrad0711 committed:
spl.py: add missing use_bootleg

Mehrad0711 committed:
bootleg.yaml: fix data dir

Mehrad0711 committed:
Only sync main data splits

Mehrad0711 committed:
default bootleg_version is master

Mehrad0711 committed:
bootleg.yaml: fix folder name

Mehrad0711 committed:
evaluate-spl: fix bug

Mehrad0711 committed:
spl.py: cleanup

Mehrad0711 committed:
Dockerfile: install genielp before bootleg to avoid dependency conflicts

Mehrad0711 committed:
bootleg.yaml: add bootleg_model name to prepped files saved path

Mehrad0711 committed:
Fix sts-paraphrase

- Remove src_lang argument. In both paraphrasing methods src and tgt lang are the same. pivot_lang is used in round_trip method only.
- Remove model_name_or_path for round_trip method since we always use MarianMT models (mbart-large-cc25 cannot be used for translation unless finetuned)

Mehrad0711 committed:
Refactor sts paraphrasing pipeline

Mehrad0711 committed:
rename return_attention to output_attentions to match genienlp

Mehrad0711 committed:
spl.py: clean up default value for args

Mehrad0711 committed:
bootleg.py: add shm volume

Mehrad0711 committed:
sts-paraphrase.yaml: fix syntax

Mehrad0711 committed:
spl.py: add --thingtatalk_column to default args

Mehrad0711 committed:
Merge branch 'master' into wip/mehrad/kubeflow

Mehrad0711 committed:
spl.py: set pull policy to always

Mehrad0711 committed:
Minor fix

Mehrad0711 committed:
Set image pull policy to always

Mehrad0711 committed:
Sync ML paraphrases to translated directory

Mehrad0711 committed:
Add 4gpu training step

Mehrad0711 committed:
Option to join ML paraphrases

Mehrad0711 committed:
Add full ML paraphrase pipeline

Mehrad0711 committed:
sts-paraphrase: rethink fewshot

Mehrad0711 committed:
Fix syntax

Mehrad0711 committed:
sts-paraphrase.yaml: fix syncing

Mehrad0711 committed:
Update Dockerfile

next images come with existing installation of genie-toolkit and thingtalk
update docker to make it compatible for both new images and old ones when using rebuild-iamge

Mehrad0711 committed:
evaluate-spl.yaml: remove unnecessary logging

Mehrad0711 committed:
Dockerfile: minor

Mehrad0711 committed:
Option to have multiple pivot_langs

Mehrad0711 committed:
Fix s3 path in sts-paraphrase

Mehrad0711 committed:
spl.py: remove default additional args

Mehrad0711 committed:
Add pipeline for masked paraphrasing on 4gpus

Mehrad0711 committed:
sts-paraphrase.yaml: sync intermediate files

Mehrad0711 committed:
Fix output path

Mehrad0711 committed:
spl.py: for each pipeline do evaluation and test in parallel

Mehrad0711 committed:
Dockerfile: remove npm linking to thingtalk

npm linking causes some dependency issues where semver is removed from thingtalk

Mehrad0711 committed:
sts-paraphrase: fix syntax

Mehrad0711 committed:
Fix spacing

gcampax commented:
I'm not sure I understand what the problem is with the existing docker file. Are you trying to use an existing image as base image for a rebuild? That doesn't make sense.

gcampax commented on pipelines/common.py:
```diff
@@ -35,7 +35,7 @@
 GENIE_VERSION = '862f3444aaee0522c84aa7b24ad0a3f7203b9f48'
 THINGTALK_VERSION = 'a3eb276cab0f554646ee6ef5620be12179f55ba7'
-BOOTLEG_VERSION = 'f53e67397ddcd099f3a18a014c9ce82b02d2223c'
+BOOTLEG_VERSION = 'master'
```
You need to use hashes in pipeline parameters, we already talked about this.

gcampax commented on sync-repos.sh:
```diff
@@ -44,7 +46,7 @@ if [ -n "${GENIE_VERSION}" ] && [ "${GENIE_VERSION}" != "${GENIE_HEAD}" ]; then
   git checkout -f ${GENIE_VERSION}
   if test -f yarn.lock ; then
-    yarn link thingtalk
+#    yarn link thingtalk
```
This line is wrong, and also should have no effect on recent versions of genie that have switched to npm. On versions of genie that use yarn, it's necessary or nothing works.

Mehrad0711 commented:
Yes. If we are always supposed to start from the base image then we won't need to change. But if you just wanna bump the version for one repo it's faster to rebuild on top of an existing image. It's just for speed but it's not needed so I can drop it.

gcampax commented:
Yeah it's better not confuse things by doing rebuilds on top of rebuilds, please drop.

Mehrad0711 committed:
Address PR comments

- Revert changes to Dockerfile used to rebuild on top of existing images
- Add back 'yarn link thingtalk' in sync-repos
- Add commit hash for bootleg
